receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -259,6 +309,7 @@ func (r *pReceiver) Shutdown(context.Context) error {
 	r.scrapeManager.Stop()
 	if r.targetAllocatorIntervalTicker != nil {
 		r.targetAllocatorIntervalTicker.Stop()
+		r.targetAllocatorStop <- true
```

Comment:
it seems like it would be safe to close the channel here, which is more idiomatic.  I don't believe shutdown is invoked more than once.